tests/test_schema.py

```diff
@@ -736,17 +758,106 @@ def test_function_ignore_parameters():
+):
+    """
+    This is a test function.
+
+    :param a: This is a parameter
+    :param b: This is another parameter
+    :param c: This is a boolean parameter
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_ignore_parameter_descriptions():
+    rf = ReferenceSchema(function_ignore_parameter_descriptions)
+    rf.remove_parameter_descriptions()
+    assert function_ignore_parameter_descriptions.schema.to_json() == rf.schema
+    assert function_ignore_parameter_descriptions.schema.to_json(SchemaType.TUNE) == rf.tune_schema
+    assert function_ignore_parameter_descriptions.tags == []
+
+
+#####################################
+#  Test ignore_function_description #
+#####################################
+
+
+@GPTEnabled(ignore_function_description=True)
+def function_ignore_function_description(a: int, b: str, c: bool = False, d: list[int] = [1, 2, 3]):
+    """
+    This is a test function.
+
+    :param a: This is a parameter
+    :param b: This is another parameter
+    :param c: This is a boolean parameter
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_ignore_function_description():
+    rf = ReferenceSchema(function_ignore_function_description)
+    rf.get_function().pop("description")
+    assert function_ignore_function_description.schema.to_json() == rf.schema
+    assert function_ignore_function_description.schema.to_json(SchemaType.TUNE) == rf.tune_schema
+    assert function_ignore_function_description.tags == []
+
+
+###############################
+#  Test ignore_all_parameters #
+###############################
+
+
+@GPTEnabled(ignore_all_parameters=True)
+def function_ignore_all_parameters(a: int, b: str, c: bool = False, d: list[int] = [1, 2, 3]):
+    """
+    This is a test function.
+
+    :param a: This is a parameter
+    :param b: This is another parameter
+    :param c: This is a boolean parameter
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_ignore_all_parameters():
+    rf = ReferenceSchema(function_ignore_all_parameters)
+    rf.get_parameters().pop("properties")
+    rf.get_parameters().pop("required")
+    assert function_ignore_all_parameters.schema.to_json() == rf.schema
+    assert function_ignore_all_parameters.schema.to_json(SchemaType.TUNE) == rf.tune_schema
+    assert function_ignore_all_parameters.tags == []
+
+
 ###############################
 #  Test global configuration  #
 ###############################
 
 
-def test_global_configuration_ignore_parameters():
-    # Change the global configuration
-    tool2schema.CONFIG.ignore_parameters = ["b", "c"]
+@pytest.fixture
+def global_config() -> tool2schema.config.Config:
+    yield tool2schema.CONFIG
+
+    # Ensure the global configuration is reset to the default after the test
+    tool2schema.CONFIG.reset_default()
+
+
+def get_locally_defined_function():
```

Comment:
Why is this needed?

Reply:
You mean the method `get_locally_defined_function` or why do we have to redefine the function?

Reply:
yeah, why do we need to call this function every time? Is it just to make sure we reset everything for each test?

Reply:
To reset the configuration we use the `global_config` fixture which automatically resets the global configuration after the test run.

`get_locally_defined_function` is used because at the moment we need to initialise `self.parameter_schemas` in the constructor, and we later rely on it in `to_json` to see if a parameter is ignored. So when you change the global configuration, nothing updates the `parameter_schemas` attribute of existing instances of FunctionSchema, and to see the change we have to redefine the function.

Now that we have switched to properties for the configuration class, we may fix this by creating a new property in FunctionSchema that returns a dictionary with the non-ignored parameters only:
```python
@property
def valid_parameter_schemas(self):
    return {k: v for k, v in self.parameter_schemas.items() if k not in self.config.ignore_parameters}
```
And this will be up to date with the current configuration.

Reply:
Yeah that sounds like a good idea.

Reply:
Done with 6a594f9